### PR TITLE
Removing deprecated labels

### PR DESCRIPTION
**Pull request by Bryce-Soghigian**
#### What type of PR is this?
Cleanup PR

#### What this PR does / why we need it:
This is a PR to remove the deprecated [labels](https://learn.microsoft.com/en-us/azure/aks/use-labels#deprecated-labels) and the tests testing those deprecated labels.
#### Which issue(s) this PR fixes:

Fixes #[2453](https://github.com/kubernetes-sigs/cloud-provider-azure/issues/2453)

#### Special notes for your reviewer:
I am wondering if we want to remove the label reconciler too since its functionality depended on those labels? or we can keep it for future labels we might want to attach to the nodes.
#### Does this PR introduce a user-facing change?

```release-note
The following labels are removed on the nodes:
* "failure-domain.beta.kubernetes.io/zone" 
* "failure-domain.beta.kubernetes.io/region"
* "beta.kubernetes.io/instance-type"
```

### CODE REVIEW

File: pkg/nodemanager/nodemanager.go [hunk 1]
```diff
@@ -72,32 +72,7 @@ var labelReconcileInfo = []struct {
...
 	secondaryKey          string
 	ensureSecondaryExists bool
-}{
-	{
-		// Reconcile the beta and the GA zone label using the beta label as
-		// the source of truth
-		// TODO: switch the primary key to GA labels in v1.21
-		primaryKey:            v1.LabelZoneFailureDomain,
-		secondaryKey:          v1.LabelZoneFailureDomainStable,
-		ensureSecondaryExists: true,
```
Comment (feiskyer) [inline pkg/nodemanager/nodemanager.go:82]: why all stable labels are removed?

Comment (Bryce-Soghigian) [reply pkg/nodemanager/nodemanager.go:82]: https://cs.github.com/kubernetes/api/blob/4e8dc44b5ed42b009e0c073c0e4a8f64d4b310ac/core/v1/well_known_labels.go#L35
It seems here all of these Stable types are also deprecated at least thats my understanding correct me if i am wrong.

Comment (Bryce-Soghigian) [reply pkg/nodemanager/nodemanager.go:82]: The  labelZoneFailureDomain and LabelZoneFailureDomainStable seem to deprecated..

Labels I removed that have deprecation announced in k8s code and release notes.
```go
		v1.LabelZoneRegion,
	v1.LabelZoneRegionStable,
	 v1.LabelZoneFailureDomain,
	v1.LabelZoneFailureDomainStable,
```
These are the zone based deprecation notices
```go
	// Retained for compat when vendored.  Do not use these consts in new code.
	LabelZoneFailureDomain       = LabelFailureDomainBetaZone   // deprecated
	LabelZoneRegion              = LabelFailureDomainBetaRegion // deprecated
	LabelZoneFailureDomainStable = LabelTopologyZone            // deprecated
	LabelZoneRegionStable        = LabelTopologyRegion          // deprecated
	
```

And for the instance types they are also announced as deprecated

```
		v1.LabelInstanceType,
	v1.LabelInstanceTypeStable
```
Have values of
```go
	LabelInstanceType       = "beta.kubernetes.io/instance-type"
	LabelInstanceTypeStable = "node.kubernetes.io/instance-type"
```
And their deprecation is announced in [here](https://learn.microsoft.com/en-us/azure/aks/use-labels#deprecated-labels)

Comment (feiskyer) [reply pkg/nodemanager/nodemanager.go:82]: The following three should be kept:

* LabelTopologyZone   = "topology.kubernetes.io/zone"
* LabelTopologyRegion = "topology.kubernetes.io/region"
* LabelInstanceTypeStable = "node.kubernetes.io/instance-type"

**Review by feiskyer — CHANGES_REQUESTED** [review]
Please ensure required labels not removed.

File: pkg/nodemanager/nodemanager.go (hunk 1 above)
Comment (Bryce-Soghigian) [reply pkg/nodemanager/nodemanager.go:82]: I added back these three labels 
```go
 map[node.kubernetes.io/instance-type:Standard_D2_v3 topology.kubernetes.io/region:eastus topology.kubernetes.io/zone:eastus-1]
```
They show up in the unit-tests and all.

File: pkg/nodemanager/nodemanager.go [hunk 2]
```diff
@@ -454,7 +429,7 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Co
 	}
 
 	// If user provided an IP address, ensure that IP address is found
-	// in the cloud provider before removing the taint on the node
+	// in the cloud provider before removing the taint 'on the node
```
Comment (feiskyer) [inline pkg/nodemanager/nodemanager.go:432]: remove `'`?

**Review by feiskyer — COMMENTED** [review]
LGTM. Could you also squash the commits after fixing the comment issue?

**Review by feiskyer — APPROVED** [review]

**Pull request closed without merging**